### PR TITLE
[ci] remove preinstalled possibly conflicting software from PATH in CI jobs

### PR DESCRIPTION
**Pull request by StrikerRUS**
With a hope to contribute to #4461.

### CODE REVIEW

File: .ci/test_r_package_windows.ps1 [hunk 1]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
 # remove some details that exist in the GitHub Actions images which might
 # cause conflicts with R and other components installed by this script
 $env:RTOOLS40_HOME = ""
+Remove-From-Path ".*Amazon.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:42]: image

File: .ci/test_r_package_windows.ps1 [hunk 2]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*Anaconda.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:43]: image

File: .ci/test_r_package_windows.ps1 [hunk 3]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*android.*"
+Remove-From-Path ".*Android.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:45]: image

File: .ci/test_r_package_windows.ps1 [hunk 4]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
 Remove-From-Path ".*chocolatey.*"
 Remove-From-Path ".*Chocolatey.*"
-Remove-From-Path ".*Git.*mingw64.*"
+Remove-From-Path ".*\\Git\\.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:48]: After cloning a repo we don't need Git in R-package jobs.
There are a lot of possible conflicts in git folder.

image

File: .ci/test_r_package_windows.ps1 [hunk 5]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*hostedtoolcache.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:49]: Strange folder with preinstalled software. I think they can install more in that folder and we'll be prepared.

image

File: .ci/test_r_package_windows.ps1 [hunk 6]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*Microsoft SDKs.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:50]: image

File: .ci/test_r_package_windows.ps1 [hunk 7]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*mingw.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:51]: For any `mingw` installation I've missed or they will install later.

File: .ci/test_r_package_windows.ps1 [hunk 8]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
 Remove-From-Path ".*msys64.*"
+Remove-From-Path ".*PostgreSQL.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:53]: image

File: .ci/test_r_package_windows.ps1 [hunk 9]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*\\R\\.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:54]: Preinstalled R itself.

image

File: .ci/test_r_package_windows.ps1 [hunk 10]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
+Remove-From-Path ".*R Client.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:55]: image

File: .ci/test_r_package_windows.ps1 [hunk 11]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
 Remove-From-Path ".*rtools40.*"
+Remove-From-Path ".*shells.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:57]: Just a strange folder.

image

File: .ci/test_r_package_windows.ps1 [hunk 12]
```diff
@@ -39,12 +39,24 @@ function Remove-From-Path {
...
 Remove-From-Path ".*Strawberry.*"
+Remove-From-Path ".*tools.*"
```
Comment (StrikerRUS) [inline .ci/test_r_package_windows.ps1:59]: Some preinstalled software. For example, there are Cygwin files in Haskell folder.

image

**Review by jameslamb — APPROVED** [review]
looks ok to me, I hope it will help!

**Pull request merged**